Task2/Partition.js

```diff
@@ -0,0 +1,23 @@
+//2- http://www.codewars.com/kata/get-the-middle-character
+function partitionOn(pred, items) {
+    var temp = [];
+    var position = 0, index = 0;
+    for (var i = 0; i < items.length; i++) {

```

Comment:
Use latest ES6 features
you can replace the old for loop with latest ES6 forEach.
use latest ES6 features whereever possible